app/src/main/java/com/lla/panicbutton/ui/fragments/onboardingfragments/ViewPagerFragment.kt

```diff
@@ -0,0 +1,79 @@
+package com.lla.panicbutton.ui.fragments.onboardingfragments
+
+import android.content.Context
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager2.widget.ViewPager2
+import com.lla.panicbutton.R
+import com.lla.panicbutton.adapters.ViewPagerAdapter
+import com.tbuonomo.viewpagerdotsindicator.SpringDotsIndicator
+import kotlinx.android.synthetic.main.fragment_view_pager.view.*
+
+class ViewPagerFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_view_pager, container, false)
+
+        val fragmentList = arrayListOf<Fragment>(
+            Welcome2Fragment(),
+            Welcome3Fragment(),
+            Welcome4Fragment()
+        )
+
+        val adapter = ViewPagerAdapter(
+            fragmentList,
+            requireActivity().supportFragmentManager,
+            lifecycle
+        )
+
+        view.viewPager2.adapter = adapter
+
+        return view
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val fragmentList = arrayListOf<Fragment>(
+            Welcome2Fragment(),
+            Welcome3Fragment(),
+            Welcome4Fragment()
+        )
+
+        val adapter = ViewPagerAdapter(
+            fragmentList,
+            requireActivity().supportFragmentManager,
+            lifecycle
+        )
+
+        view.viewPager2.adapter = adapter
+
+        val springDotsIndicator = requireView().findViewById<SpringDotsIndicator>(R.id.spring_dots_indicator)
+        val viewPager = requireView().findViewById<ViewPager2>(R.id.viewPager2)
+        springDotsIndicator?.setViewPager2(viewPager)
+
+        view.viewPagerNextButton.setOnClickListener{
+            if (viewPager.currentItem+1 < adapter.itemCount){
+                viewPager.currentItem+=1
+            }else{
+                findNavController().navigate(R.id.action_viewPagerFragment_to_setupFragment)
+                onBoardingFinished()
+            }
+        }
+    }
+
+    private fun onBoardingFinished(){
+        val sharedPref = requireActivity().getSharedPreferences("onBoarding", Context.MODE_PRIVATE)
+        val editor = sharedPref.edit()
+        editor.putBoolean("nextButton4", true)
```

Comment:
what's nextButton4? is that a name of setting? since you're putting it in the sharedPreferences.

Reply:
nextButton4 here indicates that the next button on the fourth welcome fragment is clicked